_posts/2018-06-27-inception-pipelines-pt4.markdown

```diff
@@ -0,0 +1,171 @@
+---
+layout: post
+title:  "Seeds of Inception - Part 4"
+date:   2018-06-27
+categories: aws continuous deployment codepipeline codebuild inception pipeline
+author: Pete Yandell
+image: img/inception-pipelines/seed_germination.png
+---
+
+**Part 4 seeding a forest** - A theoretical example for DevOps on AWS
+
+![forest of trees]({{ site.url }}/img/inception-pipelines/part-4-trees-lotsa-trees.png)
+
+## What's The Problem
+
+In this instalment we'll be stepping back (or is it climbing up) and looking at how Inception Pipeline fits into the forest of AWS Accounts that most organisations have. Firstly, I'll cover off what every AWS account needs (regardless of what it's used for). Next, I'll briefly discuss the common accounts that everyone should have. We'll then finish up with a few ideas on combining accounts and pipelines.
+
+## What Technologies Are We Going To Use
+
+None, nada, zero, zilch. So, put away the IDE, shutdown the AWS Console and brew up your favourite caffeinated beverage. This instalment is pure reading and thinking.
+
+## What Are The Prerequisites
+
+Not much this month. If you've read [Part 1](https://mechanicalrock.github.io/aws/continuous/deployment/2018/03/01/inception-pipelines-pt1) then that's pretty much it.
+
+## How It All Works
+
+What follows from here is my opinionated view of how Inception Pipeline fits into:
+
+* Any AWS account, no matter what the account's purpose is.
+* How you should create a suite of common / shared accounts.
+* And finally, three ideas on how you can manage something called an *aspect*.
+
+The ideas discussed in this post also fit alongside the [AWS Landing Zone](https://aws.amazon.com/answers/aws-landing-zone/) solution for managing multi-account AWS environments.
+
+**Note:** When writing this post, I struggled to find a simple, generic, way of what an Inception Pipeline does in each of the use cases below. Terms like product, feature, service, etc just didn't cover all the possible scenarios. The best term I could think up was *aspect*. Hopefully the term *aspect* makes sense (or soon will), so let’s get started!
+
+### What every AWS Account needs
+
+![account and aspect pipelines]({{ site.url }}/img/inception-pipelines/part-4-what-every-account-needs.png)
+
+The above diagram represents a very strong opinion that I have. which is that every AWS account must have at least two Inception Pipelines.
+
+The first pipeline performs actions and manages the state of the account (aka the account-level pipeline). The obvious items here are things like IAM users/groups/roles/etc, account wide KMS keys, Route 53 hosted zones, etc. The idea here is that anything that changes stuff at the account level is kept together in this pipeline. I would also say it is typically the slower changing items; created once then never really change.
```

Comment:
_The idea here is that anything that changes_ - *that* used twice in quick succession - remove one